src/blocked.c

```diff
@@ -325,7 +325,8 @@ void handleClientsBlockedOnKeys(void) {
      * (i.e. not from call(), module context, etc.) */
     serverAssert(server.also_propagate.numops == 0);
 
-    while(listLength(server.ready_keys) != 0) {
+    /* Avoid loops that constantly access the global server.ready_keys. */
```

Comment:
the comment is excessive IMHO, there's one below that tells this story.
this `if` is just to avoid excessive list creation and destruction when it's already empty.
```suggestion
```